genesyscloud/flow_loglevel/resource_genesyscloud_flow_loglevel.go

```diff
@@ -0,0 +1,158 @@
+		resources[*flowLogLevel.Id] = &resourceExporter.ResourceMeta{Name: *flowLogLevel.Id}
+	}
+
+	return resources, nil
+}
+
+// createFlowLogLevel is used by the flow_loglevel resource to create Genesyscloud flow_loglevel
+func createFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+	log.Printf("Creating flow log level for flow  %s", flowId)
+
+	flowLogLevelRequest := getFlowLogLevelSettingsRequestFromResourceData(d)
+	flowLogLevel, apiResponse, err := ep.createFlowLogLevel(ctx, flowId, &flowLogLevelRequest)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to create flow log level: %s %s", err, d.Id()), apiResponse)
+	}
+
+	log.Printf("Sucessfully created flow log level for flow:  %s flowLogLevelId: %s", flowId, *flowLogLevel.Id)
+
+	d.SetId(*flowLogLevel.Id)
+	return readFlowLogLevel(ctx, d, meta)
+}
+
+// readFlowLogLevels is used by the flow_loglevel resource to read a flow log level from genesys cloud.
+func readFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+
+	log.Printf("Reading readFlowLogLevel with flowId %s", flowId)
+	if flowId == "" {
+		log.Printf("flow log level with blank flowId %s", flowId)
+		return diag.Errorf("flowId: %s not found ", flowId)
+	}
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		flowSettingsResponse, apiResponse, err := ep.getFlowLogLevelById(ctx, flowId)
+		if err != nil {
+			log.Print(err)
+			if util.IsStatus404ByInt(apiResponse.StatusCode) {
+				return retry.NonRetryableError(fmt.Errorf("Failed to read flow log level %s: %s", flowId, err))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read flow log level %s: %s", flowId, err))
+		}
+
+		flowLogLevel := flowSettingsResponse.LogLevelCharacteristics
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceFlowLoglevel())
+		resourcedata.SetNillableValue(d, "flow_log_level", flowLogLevel.Level)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "flow_characteristics", flowLogLevel.Characteristics, flattenFlowCharacteristics)
+
+		log.Printf("Read flow log level %s", flowId)
+		return cc.CheckState()
+	})
+}
+
+// updateFlowLogLevels is used by the flow_loglevel resource to update an flow log level in Genesys Cloud
+func updateFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+	log.Printf("Updating flow log level for flow %s", flowId)
+
+	_, apiResponse, err := ep.getFlowLogLevelById(ctx, flowId)
```

Comment:
Why are you getting flow log level here?

Reply:
it's an update method so verifying that the flow exists before taking that action was the idea. I'll remove it though, seems redundant looking at it now

Reply:
I can understand why you thought this, there are a lot of update functions that do a get before the update. the reason they do this is if the genesyscloud object has a version property so we need to do a get to the the current version before updating. In this case though there's no version so you can leave it out